src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessorWorkflow.java

```diff
@@ -206,4 +238,77 @@ private List<Integer> unprocessedDocIds(final List<QuerySearchResult> querySearc
                 .collect(Collectors.toList());
         return docIds;
     }
+
+    /**
+     * This method processes a list of query top documents and retrieved the top documents
+     * that match certain criteria. If a valid top document is found, it creates and returns a sort object based on the first FieldDoc in the scoreDocs array of the top document.
+     * Otherwise, it returns null.
+     * @param queryTopDocs List of query top documents to be evaluated.
+     * @return Sort object based on the first FieldDoc in the scoreDocs array, or null if no valid document is found.
+     */
+    private Sort evaluateSortCriteria(List<CompoundTopDocs> queryTopDocs) {
+        for (CompoundTopDocs compoundTopDocs : queryTopDocs) {
+            if (compoundTopDocs != null && compoundTopDocs.getTotalHits().value > 0) {
+                List<TopDocs> topDocs = compoundTopDocs.getTopDocs();
+                TopDocs foundTopDoc = null;
+                for (TopDocs topDoc : topDocs) {
+                    if (topDoc != null && topDoc.scoreDocs.length > 0) {
+                        foundTopDoc = topDoc;
+                        break;
+                    }
+                }
+
+                if (foundTopDoc != null && foundTopDoc.scoreDocs[0] instanceof FieldDoc) {
+                    return createSort(topDocs.toArray(new TopFieldDocs[0]));
+                } else {
+                    return null;
+                }
+            }
+        }
+        return null;
+    }
```

Comment:
hard to follow the logic because of too many nested, can you refactor to limit the nesting.